src/rime/gear/corrector.cc

```diff
@@ -0,0 +1,313 @@
+//
+// Copyright RIME Developers
+// Distributed under the BSD License
+//
+// Created by nameoverflow on 2018/11/14.
+//
+
+#include <numeric>
+#include <algorithm>
+#include <queue>
+#include <rime/service.h>
+#include <rime/ticket.h>
+#include <rime/schema.h>
+#include "rime/gear/corrector.h"
```

Comment:
It seems it doesn't depend on anything in `rime/gear` and can live in `rime/algo` or `rime/dict`.
I see the use of `Prism` here. So probably the latter.